### PR TITLE
Added Wix installer native project.

### LINKED ISSUE #1276: Modifying an existing install via MSI doesn't install/uninstall any changed drivers
### Describe the bug

Modifying an existing install via MSI doesn't install/uninstall any changed drivers

### OS information

_No response_

### Steps taken to reproduce bug

* Install from MSI, leaving only the default (runtime) selected
* Run the MSI again, selecting Change, and add the "testing" component

### Expected behavior

* "sc query sampleebpfext" should show the driver running

### Actual outcome

* "sc query sampleebpfext" does not find the service
* "sc qc sampleebpfext" does not show the driver as installed

### Additional details

It is correctly installed if "testing" component was added at initial install time, but "Change" does not trigger install/uninstall of drivers.

### LINKED ISSUE #1412: MSI installer should install services with AutoStart instead of on OnDemand
### Describe the feature you'd like supported

The ebpf services are installed with `start=demand` which means if the machine reboots it they don't start back up. I encountered this after installing and rebooting due to a needed update (https://github.com/microsoft/ebpf-for-windows-demo/issues/22#issuecomment-1254266031)

### Proposed solution

Switch services to `start=auto` which mean if machine reboots ebpf starts up. I believe this would be expected behavior in most scenarios.

### Additional context

change would be

https://github.com/microsoft/ebpf-for-windows/blob/ecd08ecbb992be596090a473970afcc1eb815576/scripts/install_ebpf.psm1#L63

and maybe modify the install extension in https://github.com/microsoft/ebpf-for-windows/blob/ecd08ecbb992be596090a473970afcc1eb815576/scripts/install_ebpf.psm1#L73

### LINKED ISSUE #1478: MSI fails to install
### Describe the bug

Attempt to install the MSI

### OS information

_No response_

### Steps taken to reproduce bug

Start MSI to install. Leave default options.

### Expected behavior

Install succeeds.

### Actual outcome

Install fails. 

[install.log](https://github.com/microsoft/ebpf-for-windows/files/9824750/install.log)

```
WixQuietExec64:  WorkingDirectory is C:\Program Files\ebpf-for-windows\scripts\..
WixQuietExec64:  LogFileName is TestLog.log
WixQuietExec64:  Installing eBPF...
WixQuietExec64:   ::  OpenService FAILED 1060:
WixQuietExec64:   :: The specified service does not exist as an installed service.
WixQuietExec64:   ::  OpenService FAILED 1060:
WixQuietExec64:   :: The specified service does not exist as an installed service.
WixQuietExec64:   ::  OpenService FAILED 1060:
WixQuietExec64:   :: The specified service does not exist as an installed service.
WixQuietExec64:   ::  OpenService FAILED 1060:
WixQuietExec64:   :: The specified service does not exist as an installed service.
WixQuietExec64:   :: Element not found.
WixQuietExec64:   :: Installing NetEbpfExt...
WixQuietExec64:   ::  CreateService SUCCESS
WixQuietExec64:   :: NetEbpfExt driver created.
WixQuietExec64:   :: Installing EbpfCore...
WixQuietExec64:   ::  CreateService SUCCESS
WixQuietExec64:   :: EbpfCore driver created.
WixQuietExec64:  Failed to create eBPF user mode service.
WixQuietExec64:  At C:\Program Files\ebpf-for-windows\scripts\install_ebpf.psm1:75 char:9
WixQuietExec64:  +         throw ("Failed to create eBPF user mode service.")
WixQuietExec64:  +         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
WixQuietExec64:      + CategoryInfo          : OperationStopped: (Failed to create eBPF user mode service.:String) , RuntimeException
WixQuietExec64:      + FullyQualifiedErrorId : Failed to create eBPF user mode service.
WixQuietExec64:   
WixQuietExec64:  Error 0x80070001: Command line returned an error.
WixQuietExec64:  Error 0x80070001: QuietExec64 Failed
WixQuietExec64:  Error 0x80070001: Failed in ExecCommon method
CustomAction Install_Script returned actual error code 1603 (note this may not be 100% accurate if translation happened inside sandbox)
```

### Additional details

_No response_

### LINKED ISSUE #1629: MSI Installation issues
### Describe the bug

1. `netsh ebpf` command works first time when MSI is installed with default option (`runtime components`). If MSI is reused with say install the `Testing components` option to get the sample program files, `netsh ebpf` command stops working. -- already tracked by #1276
2. The ebpf binaries are left at multiple paths after installation is complete -- in `C:\Program Files\ebpf-for-windows` and `C:\Windows\System32`
3. eBPF store registry keys are not cleaned when the MSI is uninstalled.

### OS information

_No response_

### Steps taken to reproduce bug

**Issue1**
1. Install MSI with default option `runtime components`
2. `netsh ebpf` command should work
3. Re-run MSI and select `Testing components` option also.
4. `netsh ebpf` command stops working

### Expected behavior

NA

### Actual outcome

NA

### Additional details

_No response_

### PR DESCRIPTION
**Pull request by gtrevi**
Closes #1276
Closes #1412 (auto-starts drivers and JIT service)
Closes #1478 (will advise if the VC++ Redist is not installed)
Closes #1629
Complements work for #1883

## Description

This PR implements a new `ebpf-for-windows_installer.msi` installer, using the latest [Wix Toolset v3.11.2](https://wixtoolset.org/), and now natively integrates as a standalone and atomic project within the `ebpf-for-windows.sln` solution. It therefore replaces in full the previous installer implemented using CMake scripts with `Wix` instrumentation.

The PR also adds the following missing features and improvements:

- The entire *eBPF for Windows* installation now resides within the selected install folder only (including drivers, JIT service and `netsh` helper). Previously all `.exe` and `.dll` files were also deployed to the `\Windows\System32` folder.
- Add an additional `JIT` default sub-feature for the `eBpfSvc` (parented to the `Runtime` feature), within "Debug-only" builds.
- Add automatic installation/removal of the `sample_ebpf_ext` driver, in accordance with the installation/removal of the `Testing` feature.
- With this installer, the `eBpfSvc` service (if installed with the `JIT` feature), the `eBPFCore` and `NetEbpfExt` drivers are now automatically started once the installer finalizes the installation.

From a licensing perspective, a new `LICENSE.rtf` file was added (as Wix requires a `.RTF` format) and the installer is now customized with the [approved logo](https://github.com/microsoft/ebpf-for-windows/blob/main/docs/eBPF%20logo%20png%20800px.png) from the `ebpf-for-windows` GitHub repository.

## Testing
Manual testing, on both UI & unattended command line: Install, Change Repair and Uninstall.
Installer's feature specs:

- Runtime Components: 
    - Installs core components in the `<install folder>` and `<install folder>\drivers` folders & components
    - Adds the `<install folder>` to the system's `PATH` environment variable
    - Installs & starts the `eBpfCore` driver
    - Installs & starts the `NetEbpfExt` driver
    - `netsh` extension installed and responding to `netsh ebpf`

- JIT
    - Installs the `<install folder>\JIT` folder & components
    - Installs & starts the `eBpfSvc` service

- Development Components:
    - Installs the `<install folder>\include\*` folders & files

- Testing components
    - Installs the `<install folder>\testing` folder & components
    - Installs & starts the `SampleEbpfExt` driver

## Documentation
The following documentation was updated:

- `GettingStarted.md`
- `InstallEbpf.md`
- `LICENSE.rtf`

### CODE REVIEW

File: scripts/Setup-DevEnv.ps1
Comment (Alan-Jowett) [inline scripts/Setup-DevEnv.ps1:18]: Need to update the hash in the getting started guide as it checks to make sure this script isn't modified.

**Review by Alan-Jowett — DISMISSED** [review]

File: scripts/Setup-DevEnv.ps1
Comment (gtrevi) [reply scripts/Setup-DevEnv.ps1:18]: fixed

**Review by Alan-Jowett — DISMISSED** [review]

**Comment by Alan-Jowett** [comment]
I love the dialog box that prompts the user to install the VC redist, but can we make it a hot link? I.e. allow the user to just click the link and install instead of having to manually re-enter it?

**Comment by Alan-Jowett** [comment]
The getting started guide https://github.com/microsoft/ebpf-for-windows/blob/main/docs/GettingStarted.md says to run 
```wpr.exe -start "%ProgramFiles%\ebpf-for-windows\ebpfforwindows.wprp" -filemode``` to start tracing, but this fails because the files are in:
```"%ProgramFiles%\ebpfforwindows\ebpfforwindows.wprp"```

**Comment by gtrevi** [comment]
> The getting started guide https://github.com/microsoft/ebpf-for-windows/blob/main/docs/GettingStarted.md says to run `wpr.exe -start "%ProgramFiles%\ebpf-for-windows\ebpfforwindows.wprp" -filemode` to start tracing, but this fails because the files are in: `"%ProgramFiles%\ebpfforwindows\ebpfforwindows.wprp"`

Yes, that needs to be addressed. Open to implement what default installation folder would be best:

- `eBpfForWindows` (current)
- `ebpf-for-windows`

**Comment by dv-msft** [comment]
> > The getting started guide https://github.com/microsoft/ebpf-for-windows/blob/main/docs/GettingStarted.md says to run `wpr.exe -start "%ProgramFiles%\ebpf-for-windows\ebpfforwindows.wprp" -filemode` to start tracing, but this fails because the files are in: `"%ProgramFiles%\ebpfforwindows\ebpfforwindows.wprp"`
> 
> Yes, that needs to be addressed. Open to implement what default installation folder would be best:
> 
> * `eBpfForWindows` (current)
> * `ebpf-for-windows`

I'm assuming the user has the ability to change the install location, so the actual install location is really indeterminate, right?  Maybe we could use the default folder name in that specific command string immediately followed by an explicit note asking the user to replace that name with the name of their custom install location?

**Review by dv-msft — APPROVED** [review]

**Comment by gtrevi** [comment]
> > > The getting started guide https://github.com/microsoft/ebpf-for-windows/blob/main/docs/GettingStarted.md says to run `wpr.exe -start "%ProgramFiles%\ebpf-for-windows\ebpfforwindows.wprp" -filemode` to start tracing, but this fails because the files are in: `"%ProgramFiles%\ebpfforwindows\ebpfforwindows.wprp"`
> > 
> > 
> > Yes, that needs to be addressed. Open to implement what default installation folder would be best:
> > 
> > * `eBpfForWindows` (current)
> > * `ebpf-for-windows`
> 
> I'm assuming the user has the ability to change the install location, so the actual install location is really indeterminate, right? Maybe we could use the default folder name in that specific command string immediately followed by an explicit note asking the user to replace that name with the name of their custom install location?
> 
> In any case, IMO this can be addressed in a separate 'doc' issue.

Correct, the installation folder is completely configurable by the use (both on GUI and CLI), this is really just about the default proposed one.

File: docs/GettingStarted.md
Comment (dv-msft) [inline docs/GettingStarted.md:444]: eh... This _could_ become an irritant to the end user.  If unattended installs _are_ possible, how does it work? does the user have to confirm/supply the install location folder? If not, how would the user know what the install folder is?

Even in case of interactive installs, this would require folks to remember the install location when they actually _need_ to gather the traces (could be days after the install).  IMO a better option would be to pick a default location and hardcode it in our docs with an explicit note to change that name to the custom location.  This way, the command would work 'out-of-the-box' for folks who go with the defaults (as most would) and a helpful hint to unblock folks that go the custom route.

Comment (gtrevi) [reply docs/GettingStarted.md:444]: > eh... This _could_ become an irritant to the end user. If unattended installs _are_ possible, how does it work? does the user have to confirm/supply the install location folder? If not, how would the user know what the install folder is?

Unattended installs *are* possible, and it's documented in `InstallEbpf.md`.

> 
> Even in case of interactive installs, this would require folks to remember the install location when they actually _need_ to gather the traces (could be days after the install). IMO a better option would be to pick a default location and hardcode it in our docs with an explicit note to change that name to the custom location. This way, the command would work 'out-of-the-box' for folks who go with the defaults (as most would) and a helpful hint to unblock folks that go the custom route.

I believe that users of eBPF will have to anyways create their own scripts for different purposes within their individual contexts. IMHO, having a clear statement showing that the installation folder *is* customizable, will avoid confusion/errors from just cut & pasting from the docs.
@Alan-Jowett @dthaler @dahavey for opinions.

Comment (dv-msft) [reply docs/GettingStarted.md:444]: IMO ebpf devs (or any dev for that matter) are expected to be responsible for their _own_ scripts.  My point was about _our_ scripts and/and or suggested command lines.  If documented, IMO they _ought_ to work 'as-is' for default installs. But this is hardly an issue to hold up this PR :-)

**Review by dv-msft — APPROVED** [review]

File: docs/GettingStarted.md
Comment (gtrevi) [reply docs/GettingStarted.md:444]: As @dv-msft suggested, should we resolve this and fork this into a separate doc issue?

File: docs/InstallEbpf.md
Comment (dthaler) [inline docs/InstallEbpf.md:11]: PR description says
> This PR implements a new ebpf-for-windows_installer.msi installer, using the latest [Wix Toolset v3.11.2](https://wixtoolset.org/), and now natively integrates as a standalone and atomic project within the ebpf-for-windows.sln solution, without requiring the 'VC++ Redist' anymore (ref: https://github.com/microsoft/ebpf-for-windows/issues/1478).

So why is this step still needed in the doc?

File: scripts/run_tests.bat
Comment (dthaler) [inline scripts/run_tests.bat:6]:
```suggestion
@echo Clean up & Populate eBPF Store.
```

**Review by dthaler — APPROVED** [review]

File: docs/InstallEbpf.md
Comment (gtrevi) [reply docs/InstallEbpf.md:11]: Right, the PR description was not correct (now fixed), the VC++ Redist is required for running the eBPF binaries.

**Review by dthaler — DISMISSED** [review]

**Review by dv-msft — DISMISSED** [review]

File: docs/GettingStarted.md
Comment (gtrevi) [reply docs/GettingStarted.md:444]: Ok, I renamed the default install folder back to `ebpf-for-windows` with reference in the docs, as well as the MSI, so to match any references (i.e. recent PRs).
This way we can eventually take this topic to a separate discussion, i.e. product finalization.

**Review by saxena-anurag — COMMENTED** [review]
It looks like `Convert-BpfToNative.ps1` is not included in the MSI package. Can you please check once. This should be copied as part of `development component`.

**Comment by gtrevi** [comment]
> It looks like `Convert-BpfToNative.ps1` is not included in the MSI package. Can you please check once. This should be copied as part of `development component`.

@saxena-anurag 
Sure thing. I'm seeing that `Convert-BpfToNative.ps1` was not included in the previous installer, therefore I assumed its place could be in a new `/scripts` directory. Let me know if it'd be better placed elsewhere.

**Review by dv-msft — APPROVED** [review]

**Review by dthaler — APPROVED** [review]

**Pull request merged**
